pkgs/applications/networking/cluster/k3sup/default.nix

```diff
@@ -0,0 +1,58 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, makeWrapper
+, bash
+, openssh
+}:
+
+buildGoModule rec {
+  pname = "k3sup";
+  version = "0.11.3";
+
+  src = fetchFromGitHub {
+    owner = "alexellis";
+    repo = "k3sup";
+    rev = version;
+    sha256 = "sha256-6WYUmC2uVHFGLsfkA2EUOWmmo1dSKJzI4MEdRnlLgYY=";
+  };
+
+  buildInputs = [ makeWrapper ];
+
+  overrideModAttrs = (_: {
```

Comment:
Likely don't need to use `overrideModAttrs` here, `preBuild` during the main derivation should work as the vendor directory is writeable.